packages/block-library/src/comments-title/index.php

```diff
@@ -22,8 +22,8 @@ function render_block_core_comments_title( $attributes ) {
 	$show_post_title     = ! empty( $attributes['showPostTitle'] ) && $attributes['showPostTitle'];
 	$show_comments_count = ! empty( $attributes['showCommentsCount'] ) && $attributes['showCommentsCount'];
 	$wrapper_attributes  = get_block_wrapper_attributes( array( 'class' => $align_class_name ) );
-	$post_title          = $show_post_title ? sprintf( '"%1$s"', get_the_title() ) : null;
 	$comments_count      = number_format_i18n( get_comments_number() );
```

Comment:
```suggestion
	$comments_count      = get_comments_number();
```

For the conditions and for the `_n()` calls this needs to be in integer form to work. In Japanese this will get translated to `一` and have unexpected results below.

Reply:
I took [the code from WordPress Core](https://github.com/c4rl0sbr4v0/wordpress-develop/blob/6742d0d7a65e37f24c67eeaae373f4c086c277a5/src/wp-includes/theme-compat/comments.php#L44), and it seems that get_comments_number returns a string always.
![Screenshot 2022-05-05 at 11 26 18](https://user-images.githubusercontent.com/37012961/166896247-fd5a40d8-e929-4bf7-a65b-a2f8319351a0.png)

Is it failing on Core then, also?

Reply:
FWIW, the [docs](https://developer.wordpress.org/reference/functions/get_comments_number/#return) say

> Return
> _(string|int)_ If the post exists, a numeric string representing the number of comments the post has, otherwise 0.

Reply:
Posting this Slack message by @peterwilsoncc:

> My main concern was the translation early, as the number will get converted outside the 0-9 character set in some languages.
>
> I thought I was seeing an int when testing but that have been a numeric string.

I overall agree that using an i18n'd number (which is turned into a string that might contain non-numeric characters) seems problematic for `_n()` calls 🤔 Maybe we can err on the side of caution and do something similar as the [theme-compat code in Core](https://github.com/c4rl0sbr4v0/wordpress-develop/blob/6742d0d7a65e37f24c67eeaae373f4c086c277a5/src/wp-includes/theme-compat/comments.php#L44-L45) that @c4rl0sbr4v0 mentioned:

- Use `number_format_i18n( get_comments_number() )` only to substitute `sprintf()` placeholders
- Use `get_comments_number()` as argument in `_n()`, and in conditions.

We can continue to use the `$comments_count` variable for one of those. (I'd lean towards `number_format_i18n( get_comments_number() )` since it's longer.)

Reply:
Or maybe it's safest to drop the `$comments_count` variable altogether to make the difference a bit more explicit.

Reply:
I updated the code, putting the number format only on the sprint, but keeping `$comments_count` as a variable for comparisons and `_n` function.